index.bs

```diff
@@ -16,13 +16,441 @@ WPT Display: closed
+        1. If |params|["allowed-origin"] does not exist, return failure.
+        1. Let |allowedOrigin| be |params|["allowed-origin"].
+        1. If |allowedOrigin| is a [=structured field/token=] whose value is "`*`", return success.
+        1. If |allowedOrigin| is not a [=structured field/string=], return failure.
+        1. If the result of [=byte-serializing a request origin=] with |request| is not |allowedOrigin|'s value, then return failure.
+        1. Return success.
+    </ol>
+</div>
+
+
+<div algorithm>
+    To <dfn export>perform a storage access load check</dfn> for a [=request=] |request| and [=response=] |response|, run the following steps:
+
+    <ol class="algorithm">
+        1. Let |storageAccessStatus| be |request|'s [=request/storage access status=].
+        1. If |storageAccessStatus| is not one of "<code>[=storage access status/inactive=]</code>" or "<code>[=storage access status/active=]</code>", return failure.
+        1. Let |parsedHeader| be the result of [=header list/get a structured field value|getting a structured field value=] given "`Activate-Storage-Access`" and "`item`" from |response|'s [=response/header list=].
+        1. If |parsedHeader| is null, return failure.
+        1. Let (|value|, <var ignore>params</var>) be |parsedHeader|.
+        1. If |value| is not a [=structured field/token=], return failure.
+        1. If |value|'s value [=string/is=] not "`load`", return failure.
+        1. Return success.
+    </ol>
+</div>
+
+Integration with Fetch Metadata {#fetch-metadata-integration}
+==============================================
+
+The [:Sec-Fetch-Storage-Access:] header is appended to outgoing requests alongside other Fetch Metadata headers. [[!FETCH-METADATA]] Modify the definition of [$append the Fetch metadata headers for a request$] by inserting the following as step 6:
+
+<div algorithm="append the Sec-Fetch-Storage-Access header">
+    6.  <a lt='set-storage-access'>Set the `Sec-Fetch-Storage-Access` header</a> for <var ignore>r</var>.
+</div>
+
+Integration with Fetch {#fetch-integration}
+===========================================
+
+Handling these headers requires modifications to a few different parts of Fetch. [[!FETCH]]
+
+## `Origin` header ## {#origin-header-integration}
+
+When making a decision on whether to retry a request and force it to include unpartitioned cookies, a server ought to be informed as to the initiator of the request. I.e., the request ought to include the [:Origin:] header whenever it also includes the `Sec-Fetch-Storage-Access: inactive` header. Modify the definition of [=append a request Origin header=] by rewriting step 4 as:
+
+<div algorithm="modified append a request Origin header">
+    4. If at least one of the following conditions is true:
+         * |request|'s [=request/storage access status=] is "<code>[=storage access status/inactive=]</code>" and |request|'s [=request/credentials mode=] is "`include`"
```

Comment:
Is this condition always true _when and only when_ the `Sec-Fetch-Storage-Access` header exists? Can we just use that header's existence as the condition here, instead of repeating the condition under which that header is added, in both places?